typed-racket-lib/typed-racket/utils/any-wrap.rkt

```diff
@@ -1,21 +1,50 @@
 #lang racket/base
 
 (require racket/match racket/contract/combinator
+         racket/class racket/unit
          racket/fixnum racket/flonum racket/extflonum
          racket/set
          racket/undefined
+         (only-in racket/udp udp?)
+         (only-in racket/future future? fsemaphore?)
+         (only-in racket/pretty pretty-print-style-table?)
          (only-in (combine-in racket/private/promise)
                   promise?
                   prop:force promise-forcer))
 
 (define (base-val? e)
   (or (number? e) (string? e) (char? e) (symbol? e)
-      (null? e) (regexp? e) (eq? undefined e) (path? e)
-      (regexp? e) (keyword? e) (bytes? e) (boolean? e) (void? e)
+      (null? e) (eq? undefined e) (path? e) (eof-object? e)
+      (regexp? e) (pregexp? e) (byte-regexp? e) (byte-pregexp? e)
+      (keyword? e) (bytes? e) (boolean? e) (void? e)
+      (continuation-prompt-tag? e) (continuation-mark-key? e)
+      (module-path? e) (resolved-module-path? e)
+      (pretty-print-style-table? e) (namespace-anchor? e)
```

Comment:
I think namespace-anchors are an issue (see the typed racket issue I just filed).